src/providers/walletERC20.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  type IAgentRuntime,
+  type Provider,
+  type Memory,
+  type State,
+  type ProviderResult,
+  composePromptFromState,
+  ModelType,
+  parseKeyValueXml,
+  elizaLogger,
+} from '@elizaos/core';
+import { initWalletProvider } from './wallet';
+import { type SupportedChain } from 'src/types';
+import { searchAddressTokenTemplate } from 'src/templates';
+
+interface TokenData {
+  [chain: string]: {
+  [token: string]: {
+  tokenDecimals: number;
+  tokenAddress: string;
+  };
+  };
+  }
+
+const getTokenDecimalsAddress = async(
+  runtime: IAgentRuntime,
+  _message: Memory,
+  state?: State,
+): Promise<TokenData> =>  {
+
+  // Look through all recent messages and figure out which tokens have been mentioned and on which chains.
+  // Find the token decimals and addresses.
+  // Format in the token data field.
+
+  if (!state) {
+    state = (await runtime.composeState(_message)) as State;
```

Comment:
```suggestion
    state = await runtime.composeState(_message)
```

No need to do `as State`.  `composeState` already returns an instance of type `State`.